src/test/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipServiceTest.java

```diff
@@ -195,6 +207,42 @@ void shouldNotDeleteProgrammeMembershipWhenTraineeNotFound() {
     assertThat("Unexpected result.", result, is(false));
   }
 
+  @Test
+  void shouldSignCojWhenTraineeProgrammeMembershipFound() {
+    TraineeProfile traineeProfile = new TraineeProfile();
+    traineeProfile.getProgrammeMemberships()
+        .add(createProgrammeMembership(EXISTING_PROGRAMME_MEMBERSHIP_ID, ORIGINAL_SUFFIX, 0));
+    traineeProfile.getProgrammeMemberships()
+        .add(createProgrammeMembership("3", ORIGINAL_SUFFIX, 0));
+
+    when(repository.findByProgrammeMembershipId(EXISTING_PROGRAMME_MEMBERSHIP_ID))
+        .thenReturn(traineeProfile);
+    when(repository.save(traineeProfile)).thenAnswer(invocation -> invocation.getArgument(0));
+
+    Optional<ProgrammeMembership> programmeMembership = service
+        .signProgrammeMembershipCoj(EXISTING_PROGRAMME_MEMBERSHIP_ID);
+
+    assertThat("Unexpected optional isEmpty flag.", programmeMembership.isEmpty(), is(false));
+    assertThat("Unexpected COJ signedAt.",
+        programmeMembership.get().getConditionsOfJoining().signedAt(),
+        notNullValue());
+    assertThat("Unexpected COJ version.",
+        programmeMembership.get().getConditionsOfJoining().version(),
+        is(GoldGuideVersion.getLatest()));
+  }
+
+  @Test
+  void shouldNotSignCojWhenTraineeProgrammeMembershipNotFound() {
+    when(repository.findByProgrammeMembershipId(TRAINEE_TIS_ID)).thenReturn(null);
```

Comment:
This test only works because the default from a mock is `null`, either remove this mocked return or make sure the expectations match `randomPmId`.